pkg/asset/agent/image/oc.go

```diff
@@ -0,0 +1,133 @@
+package image
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/executer"
+	"github.com/sirupsen/logrus"
+	"github.com/thedevsaddam/retry"
+)
+
+const (
+	MachineOsImageName = "machine-os-images"
+	CoreOsFileName     = "/coreos/coreos-x86_64.iso"
+	OcDefaultTries     = 5
+	OcDefaltRetryDelay = time.Second * 5
+)
+
+type Config struct {
+	MaxTries   uint
+	RetryDelay time.Duration
+}
+
+type Release interface {
+	GetBaseIso(log logrus.FieldLogger, releaseImage string, pullSecret string, platformType models.PlatformType) (string, error)
```

Comment:
I don't see much value in using `models.PlatformType` as the type, since nothing is going to check it. Also this code is not actually agent-specific (I'd also like to use this ISO for the bootstrap in baremetal IPI), so using assisted types feels a bit odd. I'd be tempted to move this file to /pkg/coreos.

Reply:
Agree about the platform type. I'll update the PR and then investigate moving this file.